File: src/main/java/me/lutto/questuhc/listeners/ConnectListener.java
package me.lutto.questuhc.listeners;

import me.lutto.questuhc.QuestUHC;
import me.lutto.questuhc.instance.Arena;
import me.lutto.questuhc.manager.ConfigManager;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;

public class ConnectListener implements Listener {

    private QuestUHC questUHC;

    public ConnectListener(QuestUHC questUHC) {
        this.questUHC = questUHC;
    }

    @EventHandler
    public void onPlayerJoin(PlayerJoinEvent event) {
        event.getPlayer().teleport(ConfigManager.getLobbySpawn());
        event.getPlayer().setInvulnerable(true);
    }

    @EventHandler
    public void onPlayerQuit(PlayerQuitEvent event) {

        Arena arena = questUHC.getArenaManager().getArena(event.getPlayer());
        if (arena != null) {
            arena.removePlayer(event.getPlayer());
        };

    }

}

File: src/main/kotlin/me/lutto/questuhc/listeners/ConnectListener.kt
package me.lutto.questuhc.listeners

import me.lutto.questuhc.QuestUHC
import me.lutto.questuhc.manager.ConfigManager
import org.bukkit.GameMode
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent

class ConnectListener(private val questUHC: QuestUHC) : Listener {

    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) {
        event.player.teleport(ConfigManager.getLobbySpawn())
        event.player.isInvulnerable = true
        event.player.gameMode = GameMode.ADVENTURE
    }

    @EventHandler
    fun onPlayerQuit(event: PlayerQuitEvent) {
        val arena = questUHC.arenaManager.getArena(event.player)
        arena?.removePlayer(event.player)
    }

}
